### PR TITLE
Add support for custom service offerings

### PR DESCRIPTION
**Pull request by DazWorrall**
This adds 3 new params: cpu, cpu_speed, and memory, which are required together /cc @resmo

### CODE REVIEW

File: cloud/cloudstack/cs_instance.py [hunk 1]
```diff
@@ -547,6 +562,20 @@ def get_user_data(self):
...
+        cpu = self.module.params.get('cpu')
+        cpu_speed = self.module.params.get('cpu_speed')
+        memory = self.module.params.get('memory')
+        if all([cpu, cpu_speed, memory]):
+            res = [{
+                'cpuNumber': cpu,
+                'cpuSpeed': cpu_speed,
+                'memory': memory,
+            }]
+        elif any([cpu, cpu_speed, memory]):
```
Comment (resmo) [inline cloud/cloudstack/cs_instance.py]: there is a ansible module feature, where you can specify which arguments must used together, see https://github.com/DazWorrall/ansible-modules-extras/blob/feature/cs_instance_custom/cloud/cloudstack/cs_instance.py#L858. 

```
required_together = (
     ['api_key', 'api_secret', 'api_url'],
     ['cpu', 'cpu_speed', 'memory'],
),
```

**Comment by resmo** [comment]
Hi @DazWorrall thanks for this PR. 

Question, would it make sense to say: you can either pass `service_offering` or ('cpu','cpu_speed','memory') right? In this case I would also add a

```
mutually_exclusive = (
   ['cpu', 'service_offering'],
),
```

**Comment by DazWorrall** [comment]
You still need to specify a service offering that supports custom resource settings I believe - the default offering may not always do that, so I think in most cases you would need to specify both.

**Comment by resmo** [comment]
right, I see the offering has to be flagged as custom, so you can pass the custom settings. Makes sense.

LGTM. shipit @gregdek

File: cloud/cloudstack/cs_instance.py [hunk 2]
```diff
@@ -53,6 +53,21 @@
       - If not set, first found service offering is used.
     required: false
     default: null
+  cpu_count:
```
Comment (resmo) [inline cloud/cloudstack/cs_instance.py]: this should be `cpu` as you used in code

**Comment by DazWorrall** [comment]
Thanks for your time @resmo :)

File: cloud/cloudstack/cs_instance.py [hunk 3]
```diff
@@ -53,6 +53,21 @@
...
+    required: false
+    default: null
+  cpu_speed:
+    description:
+      - The clock speed/shares allocated to the instance, used with custom service offerings
+    required: false
+    default: null
+  memory:
+    description:
+      - The clock speed/shares allocated to the instance, used with custom service offerings
```
Comment (resmo) [inline cloud/cloudstack/cs_instance.py]: this seems also not quite right :)

Comment (DazWorrall) [inline cloud/cloudstack/cs_instance.py]: Yeah that's just embarassing... whoops!

**Comment by gregdek** [comment]
cc: @ansible/core for merging

**Pull request merged**